plugins/inventory/incus.py

```diff
@@ -0,0 +1,229 @@
+# Copyright (c) 2025 Stéphane Graber <stgraber@stgraber.org>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import annotations
+
+DOCUMENTATION = r"""
+name: incus
+short_description: Incus inventory source
+version_added: 12.0.0
+author:
+  - Stéphane Graber (@stgraber)
+requirements:
+  - Incus CLI (C(incus))
+description:
+  - Get inventory hosts from the Incus container and virtual-machine manager.
+options:
+  plugin:
+    description:
+      - The name of this plugin, it should always be set to community.general.incus for this plugin to work.
+    required: true
+    choices: ['community.general.incus']
+    type: str
+  default_groups:
+    description:
+      - Whether to generate default groups based on remote and project.
+    type: bool
+    default: true
+  filters:
+    description:
+      - Filter expression as supported by C(incus list).
+    type: list
+    elements: string
+    default: []
+  host_domain:
+    description:
+      - Domain to append to the host FQDN.
+    type: string
+  host_fqdn:
+    description:
+      - Whether to generate a FQDN for the host name.
+      - This will use the INSTANCE.PROJECT.REMOTE syntax.
+    type: bool
+    default: true
+  remotes:
+    description:
+      - The names of the Incus remotes to use (per C(incus remote list)).
+      - Remotes are used to access multiple servers from a single client.
+      - By default the inventory will go over all projects for each remote.
+      - It is possible to specify a specific project using V(remote:project).
+    type: list
+    elements: string
+    default: ["local"]
+extends_documentation_fragment:
+  - constructed
```

Comment:
```suggestion
  - ansible.builtin.constructed
```
(I know that other plugins also don't use FQCN here, but they should also be fixed at some point.)